[FLINK-27722] Add Slack community information and invite link (#543)
apache/flink-web: community.md

3. you keep formatting when pasting code in order to keep the code readable
4. there are enough import statements to avoid ambiguities

## Slack

> You can join the [Apache Flink community on Slack.](https://join.slack.com/t/apache-flink/shared_invite/zt-19uavkkkw-qwMimi9gawIv9Fz9xFuqrw)

Reviewer: I wonder if it makes sense to use the ASF URL shortener: https://s.apache.org/
It allows updating the underlying link (however only for the owner of the id) .. but it would still be easier than merging a change to flink-web after 100 invites.

Reviewer 2: From my side, I'd be more concerned if the invite link can only be updated by one person, compared to having to change the website once a while.

Reviewer: If that person fails to update the link, we can still update the website.
And that person should be one of the three folks involved in this PR ;)

Author: I agree with @xintongsong that I don't think that having one person who can update the shortened URL is a good idea. I rather just update the Slack invite link (for now) in the project website, since any committer can fix that.

Reviewer: Allright 👍🏻

Reviewer 3: Shall we add a note here for how to reach out to the community that the invitation link hit the limitation?

> You can join the [Apache Flink community on Slack.](https://join.slack.com/t/apache-flink/shared_invite/zt-19uavkkkw-qwMimi9gawIv9Fz9xFuqrw)

Reviewer 2: The invite link is not the only way to join the workspace. Every existing member is free to invite new members, without needing approval from admins. We may want to mention and even suggest that. Hopefully that reduces the frequency we have to update the invite link.

Author: Good point, I've added a comment for that.

After creating an account in Slack, don't forget to introduce yourself in #introductions.

> In case the invite link no longer works, please reach out to the [Dev mailing list](#mailing-Lists). 

Reviewer: ```suggestion
Due to Slack limitations the invite link expires after 100 invites. If it is expired, please reach out to the [Dev mailing list](#mailing-Lists). 
```
Maybe it makes sense to make this more explicit, so that people understand why this happens?

Reviewer 2: +1

There are a couple of community rules: 

> * **Be respectful** - This is the most important rule!

Reviewer 2: We need an empty line here, otherwise the rules are shown as a paragraph rather than a list.
```suggestion
There are a couple of community rules: 

* **Be respectful** - This is the most important rule!
```

* All important decisions and conclusions **must be reflected back to the mailing lists.** 

>   "If it didn’t happen on a mailing list, it didn’t happen." - The Apache Mottos

Reviewer: ```suggestion
  "If it didn’t happen on a mailing list, it didn’t happen." - [The Apache Mottos](http://theapacheway.com/on-list/)
```
